tdms/src/hdf5_io/hdf5_base.cpp

```diff
@@ -1,37 +1,30 @@
-#include "hdf5_io.h"
-#include "cell_coordinate.h"
+/**
+ * @file hdf5_io.cpp
+ * @authors Sam Cunliffe, William Graham
+ * @brief Common HDF5 I/O methods abstracted to the base class.
+ */
+#include "hdf5_io/hdf5_base.h"
 
-#include <exception>
 #include <iostream>
 
 #include <H5Cpp.h>
 #include <H5public.h>
 #include <spdlog/spdlog.h>
 
-/******************************************************************************
- * HDF5Writer
- */
-void HDF5Writer::write(const std::string &dataset_name, double *data, int size,
-                       hsize_t *dimensions) {
-  spdlog::debug("Writing {} to file: {}", dataset_name, filename_);
-
-  // declare a dataspace
-  H5::DataSpace dataspace(size, dimensions);
-  H5::DataType datatype(H5::PredType::NATIVE_DOUBLE);
+using namespace std;
 
-  // write the data to the dataset object in the file
-  H5::DataSet dataset = file_->createDataSet(dataset_name, datatype, dataspace);
-  dataset.write(data, H5::PredType::NATIVE_DOUBLE);
-  spdlog::trace("Write successful.");
+ijk to_ijk(const std::vector<hsize_t> dimensions) {
+  unsigned int rank = dimensions.size();
+  ijk out;
+  if (rank > 0) out.i = (int) dimensions[0];
+  if (rank > 1) out.j = (int) dimensions[1];
+  if (rank > 2) out.k = (int) dimensions[2];
+  if (rank > 3) spdlog::warn("Rank > 3");
```

Comment:
Should be a `std::logic_error` or `std::runtime_error`? Is there ever a time when we are OK with being given a 4-tensor and treating it as a 3-tensor?

Reply:
When the trailing dimensions are flat (length 1).

Although this function isn't intended for such use cases (really it's not intended for anything other than converting 3D HDF5 sizes to our native `ijk` structure that we can use throughout TBH)

Reply:
Turns out we never use this (it keeps reappearing). I'm going to remove it here.